src/recipient-recieved-scanner.ts

```diff
@@ -142,6 +170,93 @@ export class RecipientRecievedScanner {
     }
   }
 
+  /**
+   * Processes multiple distributors with error isolation.
+   * @private
+   */
+  private async processMultipleDistributors(
+    distributorsToProcess: Record<string, DistributorInfo | undefined>,
+    blockNumbersData: BlockNumberData,
+    yesterdayStr: string,
+  ): Promise<void> {
+    let successCount = 0;
+    let failureCount = 0;
+
+    // Sort distributor addresses for consistent processing order
+    const sortedAddresses = Object.keys(distributorsToProcess).sort();
+
+    for (const address of sortedAddresses) {
+      const distributorInfo = distributorsToProcess[address];
+      if (!distributorInfo) continue;
+
+      try {
+        await this.processDistributor(
+          address,
+          distributorInfo,
+          blockNumbersData,
+          yesterdayStr,
+        );
+        successCount++;
+      } catch (error) {
+        console.error(`Failed to scan distributor ${address}:`, error);
```

Comment:
Instead of catching any errors we just bubble up - any error in any distributor will cause a throw out of our class